sdk/endpoint/auth.go

```diff
@@ -0,0 +1,55 @@
+package endpoint
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/ptypes/empty"
+	"github.com/open-policy-agent/opa/rego"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/connection"
+	"github.com/networkservicemesh/networkservicemesh/controlplane/api/networkservice"
+)
+
+type AuthEndpoint struct {
+	policy rego.PreparedEvalQuery
+}
+
+func (auth *AuthEndpoint) Request(ctx context.Context, request *networkservice.NetworkServiceRequest) (*connection.Connection, error) {
+	rs, err := auth.policy.Eval(ctx, rego.EvalInput(request))
```

Comment:
I see us inputing the request... do we want to input the authentication token as well from the context?

Reply:
Yes sure, but we have to have JWT tokens merged in order to do it. And then we will be able to pass the whole chain of SPIFFE IDs